statement.go

```diff
@@ -90,3 +95,10 @@ func (stmt *snowflakeStmt) Query(args []driver.Value) (driver.Rows, error) {
 func (stmt *snowflakeStmt) GetQueryID() string {
 	return stmt.lastQueryID
 }
+
+func (stmt *snowflakeStmt) setQueryIDFromError(err error) {
+	var snowflakeError *SnowflakeError
+	if errors.As(err, &snowflakeError) {
+		stmt.lastQueryID = snowflakeError.QueryID
+	}
```

Comment:
Maybe we can add log in else here, if casting fails?

Reply:
I think that this is not necessary as there is at least 1 valid case returning regular `Error` instead of `SnowflakeError` - when the provided binding value is not supported.